lib/postgres_to_redshift.rb

```diff
@@ -81,20 +81,24 @@ def bucket
   end
 
   def copy_table(table)
-    buffer = StringIO.new
-    zip = Zlib::GzipWriter.new(buffer)
-
-    puts "Downloading #{table}"
-    copy_command = "COPY (SELECT #{table.columns_for_copy} FROM #{table.name}) TO STDOUT WITH DELIMITER '|'"
-
-    source_connection.copy_data(copy_command) do
-      while row = source_connection.get_copy_data
-        zip.write(row)
+    tmpfile = Tempfile.new("psql2rs")
+    zip = Zlib::GzipWriter.new(tmpfile)
+    begin
+      puts "Downloading #{table}"
+      copy_command = "COPY (SELECT #{table.columns_for_copy} FROM #{table.name}) TO STDOUT WITH DELIMITER '|'"
+
+      source_connection.copy_data(copy_command) do
+        while row = source_connection.get_copy_data
+          zip.write(row)
+        end
       end
+      zip.finish
+      tmpfile.rewind
+      upload_table(table, tmpfile)
+    ensure
+      zip.close
```

Comment:
Looks like I pushed my PR too soon. I should have written:

```
ensure
  zip.close unless zip.closed?
end
```
Otherwise, it causes:

```
~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/lib/postgres_to_redshift.rb:99:in `close': closed gzip stream (
Zlib::GzipFile::Error)
        from ~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/lib/postgres_to_redshift.rb:99:in `copy_table'
        from ~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/lib/postgres_to_redshift.rb:23:in `block in update_tables'
        from ~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/lib/postgres_to_redshift.rb:20:in `each'
        from ~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/lib/postgres_to_redshift.rb:20:in `update_tables'
        from ~/.gem/ruby/2.2/gems/postgres_to_redshift-0.1.1/bin/postgres_to_redshift:5:in `<top (required)>'
        from ~/bin/postgres_to_redshift:23:in `load'
        from ~/bin/postgres_to_redshift:23:in `<main>'
```